### ISSUE TITLE
allelic-mapping RNA-seq

### ISSUE THREAD

**Issue by tamerbio** [maxplanck-ie/snakepipes#523]
Hi everyone,
I am trying now to align hybrid paired-end strand-specific RNA-seq. I am using the following command:
`$RNA-seq -i /media/storage/RNA-seq_Hybrid -o /media/storage/RNA-seq_Hybrid/analysis  --local GRCm38_release98  --VCFfile /media/storage/mgp.v5.merged.snps_all.dbSNP142.vcf --strains 'CAST_EiJ,C57BL_6NJ' --mode 'allelic-mapping' --libraryType 1`
it successfully creates STAR index  but the pipeline show the following error:
```
localrules directive specifies rules that are not present in the Snakefile:
        Salmon_wasabi
        sleuth_Salmon

Building DAG of jobs...
Using shell: /bin/bash
Provided cores: 5
Rules claiming more threads will be scaled down.
Job counts:
        count   jobs
        1       all
Error in rule multiQC:
    jobid: 7
    output: multiQC/multiqc_report.html
    log: multiQC/multiQC.out, multiQC/multiQC.err (check log file(s) for error message)
    conda-env: /home/tamera/anaconda3/envs/fe60c8e03200fd2eb904733426670d5a
    shell:
        multiqc -o multiQC -f  > multiQC/multiQC.out 2> multiQC/multiQC.err
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Job failed, going on with independent jobs.
[Fri Dec  6 09:32:36 2019]
Error in rule merge_featureCounts:
    jobid: 4
    output: featureCounts/counts_allelic.tsv
    conda-env: /home/tamera/anaconda3/envs/b2fadc243261bd0cf315b10df244ac8b
    shell:
        Rscript /home/tamera/anaconda3/envs/snakePipes/lib/python3.6/site-packages/snakePipes/shared/rscripts/merge_featureCounts.R featureCounts/counts_allelic.tsv 
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Job failed, going on with independent jobs.
[Fri Dec  6 09:36:24 2019]
Finished job 1.
1 of 6 steps (17%) done

[Fri Dec  6 09:36:24 2019]
rule filter_annotation_bed:
    input: Annotation/genes.annotated.bed
    output: Annotation/genes.filtered.bed
    jobid: 2

[Fri Dec  6 09:36:25 2019]
Finished job 2.
2 of 6 steps (33%) done

[Fri Dec  6 09:36:25 2019]
rule annotation_bed2gtf:
    input: Annotation/genes.filtered.bed
    output: Annotation/genes.filtered.gtf
    jobid: 3

Activating conda environment: /home/tamera/anaconda3/envs/b2fadc243261bd0cf315b10df244ac8b
[Fri Dec  6 09:37:17 2019]
Finished job 3.
3 of 6 steps (50%) done
Exiting because a job execution failed. Look above for error message
```
i looked at the multiQC .err file and here is the content:
```
/home/tamera/anaconda3/envs/fe60c8e03200fd2eb904733426670d5a/lib/python3.6/site-packages/multiqc/utils/config.py:45: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  configs = yaml.load(f)
/home/tamera/anaconda3/envs/fe60c8e03200fd2eb904733426670d5a/lib/python3.6/site-packages/multiqc/utils/config.py:51: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  sp = yaml.load(f)
Usage: multiqc [OPTIONS] <analysis directory>

Error: Missing argument "<analysis directory>".

This is MultiQC v1.7

For more help, run 'multiqc --help' or visit http://multiqc.info


```
any idea how to solve this problem?
many many thanks?
Tamer

**Comment by katsikora** [comment]
It looks like there are several independent problems.
There seems to be a bug in the multiQC rule for the allelic-mapping mode on our side, this can be fixed.
On the other hand, do you have any trace of why featureCounts merge is failing for you?

**Comment by tamerbio** [comment]
I have no real idea .. i re-run the same command again and i got this message:
```
localrules directive specifies rules that are not present in the Snakefile:
        sleuth_Salmon
        Salmon_wasabi

Building DAG of jobs...
Using shell: /bin/bash
Provided cores: 5
Rules claiming more threads will be scaled down.
Job counts:
        count   jobs
        1       all
        1       merge_featureCounts
        1       multiQC
        3

[Mon Dec  9 15:22:04 2019]
rule merge_featureCounts:
    output: featureCounts/counts_allelic.tsv
    jobid: 4


[Mon Dec  9 15:22:04 2019]
rule multiQC:
    output: multiQC/multiqc_report.html
    log: multiQC/multiQC.out, multiQC/multiQC.err
    jobid: 7

Activating conda environment: /home/tamera/anaconda3/envs/fe60c8e03200fd2eb904733426670d5a
Activating conda environment: /home/tamera/anaconda3/envs/b2fadc243261bd0cf315b10df244ac8b
[Mon Dec  9 15:22:12 2019]
Error in rule multiQC:
    jobid: 7
    output: multiQC/multiqc_report.html
    log: multiQC/multiQC.out, multiQC/multiQC.err (check log file(s) for error message)
    conda-env: /home/tamera/anaconda3/envs/fe60c8e03200fd2eb904733426670d5a
    shell:
        multiqc -o multiQC -f  > multiQC/multiQC.out 2> multiQC/multiQC.err
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Job failed, going on with independent jobs.
NA 
Error in file(file, "rt") : cannot open the connection
Calls: lapply -> FUN -> get_df -> read.table -> file
In addition: Warning message:
In file(file, "rt") : cannot open file 'NA': No such file or directory
Execution halted
[Mon Dec  9 15:22:14 2019]
Error in rule merge_featureCounts:
    jobid: 4
    output: featureCounts/counts_allelic.tsv
    conda-env: /home/tamera/anaconda3/envs/b2fadc243261bd0cf315b10df244ac8b
    shell:
        Rscript /home/tamera/anaconda3/envs/snakePipes/lib/python3.6/site-packages/snakePipes/shared/rscripts/merge_featureCounts.R featureCounts/counts_allelic.tsv 
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Job failed, going on with independent jobs.
Exiting because a job execution failed. Look above for error message
Complete log: /media/storage/RNA-seq_Hybrid/analysis/.snakemake/log/2019-12-09T152203.649113.snakemake.log

 !!! ERROR in RNA-seq workflow! !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Error: snakemake returned an error code of 1, so processing is incomplete!
```

**Comment by tamerbio** [comment]
are  `multiQC` and `featureCounts`  complete independent task in the pipeline?

**Comment by katsikora** [comment]
featureCounts is independent of multiQC. MultiQC will eventually attempt to use featureCounts output, but in your log is failing (beforehand) because of a missing argument to -f. That's a bug on our side, I'll  look into it.

**Comment by katsikora** [comment]
Hi there,

I cannot confirm a problem with featureCounts_merge in allelic RNAseq for snakePipes version 1.3.0, it works for our test data. 
It looks like there is a problem with matching sample or genotype names in your case ('NA'). What does the content of your featureCounts folder, but also snp_genome folder look like? 

MultiQC is a real bug, I'll make a fix.

**Comment by tamerbio** [comment]
Is there any possible way to perform the allele-specific alignment manually until the second release of your great tool is out???

**Comment by tamerbio** [comment]
my snp_genome folder is:
```
-rw-rwxr--+  1 tamera users 724352171 Dec  4 18:32 all_C57BL_6NJ_SNPs_CAST_EiJ_reference.based_on_GRCm38_release98.txt*
-rw-rwxr--+  1 tamera users    160342 Dec  4 20:03 all_SNPs_C57BL_6NJ_GRCm38.txt.gz*
-rw-rwxr--+  1 tamera users 233326688 Dec  4 15:37 all_SNPs_CAST_EiJ_GRCm38.txt.gz*
drwxrwxr-x+  2 tamera users         0 Dec  4 20:04 C57BL_6NJ_full_sequence/
-rw-rwxr--+  1 tamera users      3130 Dec  4 20:04 C57BL_6NJ_genome_preparation_report.txt*
drwxrwxr-x+  2 tamera users         0 Dec  4 20:04 C57BL_6NJ_N-masked/
-rw-rwxr--+  1 tamera users       694 Dec  4 18:32 C57BL_6NJ_SNP_filtering_report.txt*
-rw-rwxr--+  1 tamera users     79121 Dec  4 20:08 C57BL_6NJ_specific_SNPs.GRCm38_release98.txt*
drwxrwxr-x+  2 tamera users         0 Dec  4 20:13 CAST_EiJ_C57BL_6NJ_dual_hybrid.based_on_GRCm38_release98_full_sequence/
drwxrwxr-x+  2 tamera users         0 Nov 29 18:58 CAST_EiJ_C57BL_6NJ_dual_hybrid.based_on_GRCm38_release98_N-masked/
-rw-rwxr--+  1 tamera users      4011 Dec  4 20:13 CAST_EiJ_C57BL_6NJ_dual_hybrid.genome_preparation_report.txt*
-rw-rwxr--+  1 tamera users    194086 Dec  4 20:08 CAST_EiJ_C57BL_6NJ_SNPs_in_common.GRCm38_release98.txt*
drwxrwxr-x+  2 tamera users         0 Dec  4 15:44 CAST_EiJ_full_sequence/
-rw-rwxr--+  1 tamera users      3270 Dec  4 15:44 CAST_EiJ_genome_preparation_report.txt*
drwxrwxr-x+  2 tamera users         0 Dec  4 15:44 CAST_EiJ_N-masked/
-rw-rwxr--+  1 tamera users       699 Dec  4 18:32 CAST_EiJ_SNP_filtering_report.txt*
-rw-rwxr--+  1 tamera users 322737589 Dec  4 20:08 CAST_EiJ_specific_SNPs.GRCm38_release98.txt*
drwxrwxr-x+  2 tamera users         0 Dec  4 15:44 SNPs_C57BL_6NJ/
drwxrwxr-x+  2 tamera users         0 Dec  4 11:49 SNPs_CAST_EiJ/
-rw-rwxr--+  1 tamera users     60404 Dec  4 20:13 SNPsplit_createSNPgenome.err*
-rw-rwxr--+  1 tamera users      2320 Dec  4 20:13 SNPsplit_createSNPgenome.out*
drwxrwxr-x+  2 tamera users         0 Dec  5 14:23 star_Nmasked/

```


my featureCounts folder is empty

**Comment by katsikora** [comment]
Ok... so the genome generation step went through ok in your analysis. So the genome and sample name matching is working fine.
Something failed before featureCounts merge.
In the folder cluster_logs, can you find any error in files with featureCounts_allele* or snp_split*?
Do you find bam files split by genome1 and genome2 for each of your samples in the folder allelic_bams ?

**Comment by tamerbio** [comment]
but why `C57BL_6NJ_full_sequence` has duplicated chromosome files??

**Comment by katsikora** [comment]
Right, I don't even know how that's possible. What kind of system are you running on?
This doesn't happen when I run the workflow on our example data. We have CentOS7.

**Comment by tamerbio** [comment]
I am running snakePipe on Ubuntu 14.04.6 LTS

**Comment by tamerbio** [comment]
I am trying from the beginning all over again … Let me see if get the same output !

**Comment by tamerbio** [comment]
Hi @katsikora ,
i re-run the command again.. i have got the same error message.. but this time snp_genome seems to be fine:
├── Annotation
│   ├── filter_command.txt
│   ├── genes.annotated.bed
│   ├── genes.filtered.bed
│   └── genes.filtered.gtf
├── cluster_logs
├── featureCounts
├── Log.out
├── multiQC
│   ├── multiQC.err
│   └── multiQC.out
├── RNA-seq.cluster_config.yaml
│   │   ├── chr12.SNPs_introduced.fa
│   │   ├── chr13.SNPs_introduced.fa
│   │   ├── chr14.SNPs_introduced.fa
│   │   ├── chr15.SNPs_introduced.fa
│   │   ├── chr16.SNPs_introduced.fa
│   │   ├── chr17.SNPs_introduced.fa
│   │   ├── chr18.SNPs_introduced.fa
│   │   ├── chr19.SNPs_introduced.fa
│   │   ├── chr1.SNPs_introduced.fa
│   │   ├── chr2.SNPs_introduced.fa
│   │   ├── chr3.SNPs_introduced.fa
│   │   ├── chr4.SNPs_introduced.fa
│   │   ├── chr5.SNPs_introduced.fa
│   │   ├── chr6.SNPs_introduced.fa
│   │   ├── chr7.SNPs_introduced.fa
│   │   ├── chr8.SNPs_introduced.fa
│   │   ├── chr9.SNPs_introduced.fa
│   │   ├── chrMT.SNPs_introduced.fa
│   │   ├── chrX.SNPs_introduced.fa
│   │   └── chrY.SNPs_introduced.fa
│   ├── C57BL_6NJ_genome_preparation_report.txt
│   ├── C57BL_6NJ_N-masked
│   │   ├── chr10.N-masked.fa
│   │   ├── chr11.N-masked.fa
│   │   ├── chr12.N-masked.fa
│   │   ├── chr13.N-masked.fa
│   │   ├── chr14.N-masked.fa
│   │   ├── chr15.N-masked.fa
│   │   ├── chr16.N-masked.fa
│   │   ├── chr17.N-masked.fa
│   │   ├── chr18.N-masked.fa
│   │   ├── chr19.N-masked.fa
│   │   ├── chr1.N-masked.fa
│   │   ├── chr2.N-masked.fa
│   │   ├── chr3.N-masked.fa
│   │   ├── chr4.N-masked.fa
│   │   ├── chr5.N-masked.fa
│   │   ├── chr6.N-masked.fa
│   │   ├── chr7.N-masked.fa
│   │   ├── chr8.N-masked.fa
│   │   ├── chr9.N-masked.fa
│   │   ├── chrMT.N-masked.fa
│   │   ├── chrX.N-masked.fa
│   │   └── chrY.N-masked.fa
│   ├── C57BL_6NJ_SNP_filtering_report.txt
│   ├── C57BL_6NJ_specific_SNPs.GRCm38_release98.txt
│   ├── CAST_EiJ_C57BL_6NJ_dual_hybrid.based_on_GRCm38_release98_full_sequence
│   │   ├── chr10.SNPs_introduced.fa
│   │   ├── chr11.SNPs_introduced.fa
│   │   ├── chr12.SNPs_introduced.fa
│   │   ├── chr13.SNPs_introduced.fa
│   │   ├── chr14.SNPs_introduced.fa
│   │   ├── chr15.SNPs_introduced.fa
│   │   ├── chr16.SNPs_introduced.fa
│   │   ├── chr17.SNPs_introduced.fa
│   │   ├── chr18.SNPs_introduced.fa
│   │   ├── chr19.SNPs_introduced.fa
│   │   ├── chr1.SNPs_introduced.fa
│   │   ├── chr2.SNPs_introduced.fa
│   │   ├── chr3.SNPs_introduced.fa
│   │   ├── chr4.SNPs_introduced.fa
│   │   ├── chr5.SNPs_introduced.fa
│   │   ├── chr6.SNPs_introduced.fa
│   │   ├── chr7.SNPs_introduced.fa
│   │   ├── chr8.SNPs_introduced.fa
│   │   ├── chr9.SNPs_introduced.fa
│   │   ├── chrMT.SNPs_introduced.fa
│   │   ├── chrX.SNPs_introduced.fa
│   │   └── chrY.SNPs_introduced.fa
│   ├── CAST_EiJ_C57BL_6NJ_dual_hybrid.based_on_GRCm38_release98_N-masked
│   │   ├── chr10.N-masked.fa
│   │   ├── chr11.N-masked.fa
│   │   ├── chr12.N-masked.fa
│   │   ├── chr13.N-masked.fa
│   │   ├── chr14.N-masked.fa
│   │   ├── chr15.N-masked.fa
│   │   ├── chr16.N-masked.fa
│   │   ├── chr17.N-masked.fa
│   │   ├── chr18.N-masked.fa
│   │   ├── chr19.N-masked.fa
│   │   ├── chr1.N-masked.fa
│   │   ├── chr2.N-masked.fa
│   │   ├── chr3.N-masked.fa
│   │   ├── chr4.N-masked.fa
│   │   ├── chr5.N-masked.fa
│   │   ├── chr6.N-masked.fa
│   │   ├── chr7.N-masked.fa
│   │   ├── chr8.N-masked.fa
│   │   ├── chr9.N-masked.fa
│   │   ├── chrMT.N-masked.fa
│   │   ├── chrX.N-masked.fa
│   │   └── chrY.N-masked.fa
│   ├── CAST_EiJ_C57BL_6NJ_dual_hybrid.genome_preparation_report.txt
│   ├── CAST_EiJ_C57BL_6NJ_SNPs_in_common.GRCm38_release98.txt
│   ├── CAST_EiJ_full_sequence
│   │   ├── chr10.SNPs_introduced.fa
│   │   ├── chr11.SNPs_introduced.fa
│   │   ├── chr12.SNPs_introduced.fa
│   │   ├── chr13.SNPs_introduced.fa
│   │   ├── chr14.SNPs_introduced.fa
│   │   ├── chr15.SNPs_introduced.fa
│   │   ├── chr16.SNPs_introduced.fa
│   │   ├── chr17.SNPs_introduced.fa
│   │   ├── chr18.SNPs_introduced.fa
│   │   ├── chr19.SNPs_introduced.fa
│   │   ├── chr1.SNPs_introduced.fa
│   │   ├── chr2.SNPs_introduced.fa
│   │   ├── chr3.SNPs_introduced.fa
│   │   ├── chr4.SNPs_introduced.fa
│   │   ├── chr5.SNPs_introduced.fa
│   │   ├── chr6.SNPs_introduced.fa
│   │   ├── chr7.SNPs_introduced.fa
│   │   ├── chr8.SNPs_introduced.fa
│   │   ├── chr9.SNPs_introduced.fa
│   │   ├── chrMT.SNPs_introduced.fa
│   │   ├── chrX.SNPs_introduced.fa
│   │   └── chrY.SNPs_introduced.fa
│   ├── CAST_EiJ_genome_preparation_report.txt
│   ├── CAST_EiJ_N-masked
│   │   ├── chr10.N-masked.fa
│   │   ├── chr11.N-masked.fa
│   │   ├── chr12.N-masked.fa
│   │   ├── chr13.N-masked.fa
│   │   ├── chr14.N-masked.fa
│   │   ├── chr15.N-masked.fa
│   │   ├── chr11.txt
│   │   ├── chr12.txt
│   │   ├── chr13.txt
│   │   ├── chr14.txt
│   │   ├── chr15.txt
│   │   ├── chr16.txt
│   │   ├── chr17.txt
│   │   ├── chr18.txt
│   │   ├── chr19.txt
│   │   ├── chr1.txt
│   │   ├── chr2.txt
│   │   ├── chr3.txt
│   │   ├── chr4.txt
│   │   ├── chr5.txt
│   │   ├── chr6.txt
│   │   ├── chr7.txt
│   │   ├── chr8.txt
│   │   ├── chr9.txt
│   │   ├── chrMT.txt
│   │   ├── chrX.txt
│   │   └── chrY.txt
│   ├── SNPs_CAST_EiJ
│   │   ├── chr10.txt
│   │   ├── chr11.txt
│   │   ├── chr12.txt
│   │   ├── chr13.txt
│   │   ├── chr14.txt
│   │   ├── chr15.txt
│   │   ├── chr16.txt
│   │   ├── chr17.txt
│   │   ├── chr18.txt
│   │   ├── chr19.txt
│   │   ├── chr1.txt
│   │   ├── chr2.txt
│   │   ├── chr3.txt
│   │   ├── chr4.txt
│   │   ├── chr5.txt
│   │   ├── chr6.txt
│   │   ├── chr7.txt
│   │   ├── chr8.txt
│   │   ├── chr9.txt
│   │   ├── chrMT.txt
│   │   ├── chrX.txt
│   │   └── chrY.txt
│   ├── SNPsplit_createSNPgenome.err
│   ├── SNPsplit_createSNPgenome.out
│   └── star_Nmasked
│       ├── chrLength.txt
│       ├── chrNameLength.txt
│       ├── chrName.txt
│       ├── chrStart.txt
│       ├── exonGeTrInfo.tab
│       ├── exonInfo.tab
│       ├── geneInfo.tab
│       ├── Genome
│       ├── genomeParameters.txt
│       ├── SA
│       ├── SAindex
│       ├── sjdbInfo.txt
│       ├── sjdbList.fromGTF.out.tab
│       ├── sjdbList.out.tab
│       ├── star.index.err
│       ├── star.index.out
│       └── transcriptInfo.tab
├── SNPsplit_createSNPgenome.err
└── SNPsplit_createSNPgenome.out

**Comment by katsikora** [comment]
Hi Tamerbio,

can you check this:
In the folder cluster_logs, can you find any error in files with featureCounts_allele* or snp_split*?
Do you find bam files split by genome1 and genome2 for each of your samples in the folder allelic_bams ?

**Comment by tamerbio** [comment]
RNA-seq log file:
```
/home/tamera/anaconda3/envs/snakePipes/bin/RNA-seq -i /media/storage/RNA-seq_Hybrid -o /media/storage/RNA-seq_Hybrid/analysis --local GRCm38_release98 --VCFfile /media/storage/mgp.v5.merged.snps_all.dbSNP142.vcf --strains CAST_EiJ,C57BL_6NJ --mode allelic-mapping --libraryType 1

TMPDIR=/media/storage/PYTHONNOUSERSITE=True /home/tamera/anaconda3/envs/snakePipes/bin/snakemake --use-conda --conda-prefix /home/tamera/anaconda3/envs --latency-wait 300 --snakefile /home/tamera/anaconda3/envs/snakePipes/lib/python3.6/site-packages/snakePipes/workflows/RNA-seq/Snakefile --jobs 5 --directory /media/storage/RNA-seq_Hybrid/analysis --configfile /media/storage/RNA-seq_Hybrid/analysis/RNA-seq.config.yaml --keep-going

localrules directive specifies rules that are not present in the Snakefile:
	sleuth_Salmon
	Salmon_wasabi

Building DAG of jobs...
Using shell: /bin/bash
Provided cores: 5
Rules claiming more threads will be scaled down.
Job counts:
	count	jobs
	1	all
	1	annotation_bed2gtf
	1	create_annotation_bed
	1	create_snpgenome
	1	filter_annotation_bed
	1	merge_featureCounts
	1	multiQC
	1	star_index
	8

[Tue Dec 10 13:18:07 2019]
rule merge_featureCounts:
    output: featureCounts/counts_allelic.tsv
    jobid: 4


[Tue Dec 10 13:18:07 2019]
rule multiQC:
    output: multiQC/multiqc_report.html
    log: multiQC/multiQC.out, multiQC/multiQC.err
    jobid: 7


[Tue Dec 10 13:18:07 2019]
rule create_annotation_bed:
    input: /media/storage/resSnakePipe/annotation/genes.gtf
    output: Annotation/genes.annotated.bed
    jobid: 1


[Tue Dec 10 13:18:07 2019]
rule create_snpgenome:
    input: /media/Astorage/resSnakePipe/genome_fasta
    output: snp_genome/CAST_EiJ_SNP_filtering_report.txt, snp_genome/C57BL_6NJ_SNP_filtering_report.txt, snp_genome/CAST_EiJ_C57BL_6NJ_dual_hybrid.based_on_GRCm38_release98_N-masked, snp_genome/all_C57BL_6NJ_SNPs_CAST_EiJ_reference.based_on_GRCm38_release98.txt
    log: SNPsplit_createSNPgenome.out, SNPsplit_createSNPgenome.err
    jobid: 5

Activating conda environment: /home/tamera/anaconda3/envs/fe60c8e03200fd2eb904733426670d5a
Activating conda environment: /home/tamera/anaconda3/envs/b2fadc243261bd0cf315b10df244ac8b
Activating conda environment: /home/tamera/anaconda3/envs/b2fadc243261bd0cf315b10df244ac8b
Activating conda environment: /home/tamera/anaconda3/envs/fe60c8e03200fd2eb904733426670d5a
[Tue Dec 10 13:18:10 2019]
Error in rule multiQC:
    jobid: 7
    output: multiQC/multiqc_report.html
    log: multiQC/multiQC.out, multiQC/multiQC.err (check log file(s) for error message)
    conda-env: /home/tamera/anaconda3/envs/fe60c8e03200fd2eb904733426670d5a
    shell:
        multiqc -o multiQC -f  > multiQC/multiQC.out 2> multiQC/multiQC.err
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Job failed, going on with independent jobs.
NA 
Error in file(file, "rt") : cannot open the connection
Calls: lapply -> FUN -> get_df -> read.table -> file
In addition: Warning message:
In file(file, "rt") : cannot open file 'NA': No such file or directory
Execution halted
[Tue Dec 10 13:18:13 2019]
Error in rule merge_featureCounts:
    jobid: 4
    output: featureCounts/counts_allelic.tsv
    conda-env: /home/tamera/anaconda3/envs/b2fadc243261bd0cf315b10df244ac8b
    shell:
        Rscript /home/tamera/anaconda3/envs/snakePipes/lib/python3.6/site-packages/snakePipes/shared/rscripts/merge_featureCounts.R featureCounts/counts_allelic.tsv 
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Job failed, going on with independent jobs.
[Tue Dec 10 13:22:26 2019]
Finished job 1.
1 of 8 steps (12%) done

[Tue Dec 10 13:22:26 2019]
rule filter_annotation_bed:
    input: Annotation/genes.annotated.bed
    output: Annotation/genes.filtered.bed
    jobid: 2

[Tue Dec 10 13:22:27 2019]
Finished job 2.
2 of 8 steps (25%) done

[Tue Dec 10 13:22:27 2019]
rule annotation_bed2gtf:
    input: Annotation/genes.filtered.bed
    output: Annotation/genes.filtered.gtf
    jobid: 3

Activating conda environment: /home/tamera/anaconda3/envs/b2fadc243261bd0cf315b10df244ac8b
[Tue Dec 10 13:23:20 2019]
Finished job 3.
3 of 8 steps (38%) done
[Tue Dec 10 21:30:59 2019]
Finished job 5.
4 of 8 steps (50%) done

[Tue Dec 10 21:30:59 2019]
rule star_index:
    input: snp_genome/CAST_EiJ_C57BL_6NJ_dual_hybrid.based_on_GRCm38_release98_N-masked
    output: snp_genome/star_Nmasked/Genome
    log: snp_genome/star_Nmasked/star.index.out, snp_genome/star_Nmasked/star.index.err
    jobid: 6
    threads: 5

Activating conda environment: /home/tamera/anaconda3/envs/b2fadc243261bd0cf315b10df244ac8b
[Tue Dec 10 22:21:45 2019]
Finished job 6.
5 of 8 steps (62%) done
Exiting because a job execution failed. Look above for error message
Complete log: /media/ATLAS_NGS_storage/Phillip_Grote/Tamer/Hybrid/RNA-seq_Hybrid/analysis/.snakemake/log/2019-12-10T131806.857214.snakemake.log

 !!! ERROR in RNA-seq workflow! !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!


```

**Comment by tamerbio** [comment]
`cluster_logs `folder is empty and no Bam file at all .. the workflow is broken before the alignment step even started

**Comment by katsikora** [comment]
Looks very weird. Mapping is not listed among the jobs the workflow has to execute.

An example workflow including all the necessary steps looks like this:

Job counts:
	count	jobs
	40	BAMindex_allelic
	40	BAMsort_allelic
	10	FASTQ1
	10	FASTQ2
	20	FastQC
	20	FastQC_on_trimmed
	10	STAR_allele
	1	all
	1	annotation_bed2gtf
	20	bamCoverage_RPKM_allelic
	1	create_annotation_bed
	1	create_snpgenome
	10	cutadapt
	10	featureCounts_allele
	1	filter_annotation_bed
	1	merge_featureCounts
	40	movebams
	1	multiQC
	10	origFASTQ1
	10	origFASTQ2
	10	sambamba_markdup
	10	snp_split
	1	star_index
	278

**Comment by tamerbio** [comment]
here is my command:
`RNA-seq -i /media/storage/RNA-seq_Hybrid -o /media/storage/RNA-seq_Hybrid/analysis  --local GRCm38_release98  --VCFfile /media/storage/mgp.v5.merged.snps_all.dbSNP142.vcf --strains 'CAST_EiJ,C57BL_6NJ' --mode  allelic-mapping  --libraryType 1`

**Comment by LeilyR** [comment]
Try a new (and definitely an empty) path as your output path.  I suspect that some files already exist in your output folder that pipeline detects them and that is why they don't get listed in your dag.
Also you would never get a cluster log when you run the pipeline locally (--local)

**Comment by katsikora** [comment]
The command looks fine.

Maybe the reads could not be identified?
What is in your input folder? Do you have reads there directly, or in some subfolders? What are the read extentions?

**Comment by tamerbio** [comment]
the input directory contains:
├── hyb1_R1.fq
├── hyb1_R1.fq.gz
├── hyb1_R2.fq
├── hyb1_R2.fq.gz
├── hyb3_R1.fq
├── hyb3_R1.fq.gz
├── hyb3_R2.fq
├── hyb3_R2.fq.gz
├── hyb4_R1.fq
├── hyb4_R1.fq.gz

**Comment by tamerbio** [comment]
@LeilyR I tried already new output directory but the same job counts as mentioned above

**Comment by katsikora** [comment]
Ok, so this mix of 'fq' and 'fq.gz' might be confusing read detection.
On top of this, the default extention the workflow is looking for is  '.fastq.gz'.
This can be changed by modifying the 'ext' entry of the config.yaml, a copy of which you should have in your workflow output directory.

To get this to work:
1)please remove the the *fq files from the input folder or move them somewhere else
2)either rename your files to have the extention 'fastq.gz' instead of 'fq.gz' or add  --ext 'fq.gz' to your commandline.

**Comment by tamerbio** [comment]
seems working ... thanks a ton .. it is running now and i keep you posted with the results

**Comment by katsikora** [comment]
The fix to multiQC and checks for successful sample identification are now available in the release 1.3.2.
